remotes/docker/config/hosts.go

```diff
@@ -57,10 +60,11 @@ type hostConfig struct {
 
 // HostOptions is used to configure registry hosts
 type HostOptions struct {
-	HostDir       func(string) (string, error)
-	Credentials   func(host string) (string, string, error)
-	DefaultTLS    *tls.Config
-	DefaultScheme string
+	HostDir          func(string) (string, error)
+	Credentials      func(host string) (string, string, error)
+	DefaultTLS       *tls.Config
+	DefaultScheme    string
+	TransportWrapper TransportWrapperFunc
```

Comment:
Rather than accepting a `transportWrapper`, I think it might be more flexible for the resolver to accept a function that mutates an `http.Client` as an option.  You could then pass the generated default `http.Client` to that function to get modified however the caller wishes.

```go
type HTTPClientOptionFunc func(*http.Client) error
```

```go
client := &http.Client{
	Transport: defaultTransport
}
if options.HTTPClientOption != nil {
	if err := options.HTTPClientOption(client); err != nil {
		return nil, err
	}
}
```